app/components/fixup_embeds.py

```diff
@@ -20,26 +20,28 @@
 
     from app.bot import GhosttyBot
 
-
-# Ignore `)` if at the end of the link
+VALID_HTTP_CHARS = r"[A-Za-z0-9-._~:/?#\[\]@!$&'()*+,;%=]"
 EMBED_SITES: tuple[tuple[re.Pattern[str], Callable[[re.Match[str]], str]], ...] = (
     (
-        re.compile(
-            r"https://(?:www\.)?(?:x|twitter)\.com/"
-            r"([A-Za-z0-9-._~:/?#\[\]@!$&'()*+,;%=]+)\)?"
-        ),
+        re.compile(rf"https://(?:www\.)?x\.com/({VALID_HTTP_CHARS}+)"),
         lambda match: f"https://fixupx.com/{match[1]}",
     ),
     (
-        re.compile(
-            r"https://(?:www\.)?bsky\.app/([A-Za-z0-9-._~:/?#\[\]@!$&'()*+,;%=]+)\)?"
-        ),
+        re.compile(rf"https://(?:www\.)?twitter\.com/({VALID_HTTP_CHARS}+)"),
+        lambda match: f"https://fxtwitter.com/{match[1]}",
+    ),
+    (
+        re.compile(rf"https://(?:www\.)?bsky\.app/({VALID_HTTP_CHARS}+)"),
         lambda match: f"https://fxbsky.app/{match[1]}",
     ),
+    (
+        re.compile(rf"https://(?:www\.)?pixiv\.net/({VALID_HTTP_CHARS}+)"),
+        lambda match: f"https://phixiv.net/{match[1]}",
+    ),
     (
         re.compile(
-            r"https://(?:(?:www|(?P<subreddit>\w+))?\.)?reddit\.com/"
-            r"(?P<extra>[A-Za-z0-9-._~:/?#\[\]@!$&'()*+,;%=]+)\)?"
+            r"https://(?:(?:www|(?P<subreddit>\w+))\.)?reddit\.com/"
+            rf"(?P<extra>{VALID_HTTP_CHARS}+)"
```

Comment:
```suggestion
            rf"(?P<post>{VALID_HTTP_CHARS}+)"
```

It's not really “extra”: it's the main part.